packages/@react-native-windows/cli/src/commands/autolinkWindows/autolinkWindows.ts

```diff
@@ -284,23 +278,51 @@ export class AutoLinkWindows {
     });
   }
 
+  private getAutolinkTemplateFile(
+    templateRoot: string,
+    targetFile: string,
+  ): string {
+    if (templateRoot.endsWith('\\template')) {
+      // Old template split into different folders
+      switch (path.extname(targetFile)) {
+        case '.cpp':
+        case '.h':
+          templateRoot = path.join(templateRoot, 'cpp-app', 'src');
+          break;
+        case '.cs':
+          templateRoot = path.join(templateRoot, 'cs-app', 'src');
+          break;
+        case '.props':
+        case '.targets':
+        default:
+          templateRoot = path.join(templateRoot, 'shared-app', 'src');
```

Comment:
Just curious, what is a `shared-app`?

Reply:
The old paper templates were grouped into sets of folders based on type (shared by all apps, shared by all libs, project vs source, etc) so they could be re-used by multiple templates. However this meant a bunch of custom logic in the generator to decide which files to use in which situations.

With `init-windows`, the decision was to simplify the matrix of options into named templates, where you could open a single folder with its literal name and see exactly which files it was going to copy in a new project.